src/main/java/io/cdap/plugin/http/source/common/pagination/page/JsonPage.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.http.source.common.pagination.page;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import io.cdap.plugin.http.source.common.BaseHttpSourceConfig;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Iterator;
+import java.util.Map;
+
+/**
+ * Returns elements from json one by one by given json path.
+ */
+class JsonPage implements BasePage {
+  private final String insideElementJsonPathPart;
+  private final Iterator<JsonElement> iterator;
+  private final JsonObject json;
+  private final Map<String, String> fieldsMapping;
+
+  JsonPage(BaseHttpSourceConfig config, String responseBody) {
+    json = JSONUtil.toJsonObject(responseBody);
+    JSONUtil.JsonQueryResponse queryResponse = JSONUtil.getJsonElementByPath(json, config.getResultPath());
+    insideElementJsonPathPart = queryResponse.getUnretrivedPath();
+    iterator = queryResponse.getAsJsonArray().iterator();
+    fieldsMapping = config.getFullFieldsMapping();
+  }
+
+  @Override
+  public boolean hasNext() {
+    return iterator.hasNext();
+  }
+
+  @Override
+  public String next() {
+    JsonObject currentJsonObject = iterator.next().getAsJsonObject();
+
+    JsonObject resultJson = new JsonObject();
+    for (Map.Entry<String, String> entry : fieldsMapping.entrySet()) {
+      String schemaFieldName = entry.getKey();
+      String fieldPath = insideElementJsonPathPart + "/" + StringUtils.stripStart(entry.getValue(), "/");
+
+      JSONUtil.JsonQueryResponse queryResponse = JSONUtil.getJsonElementByPath(currentJsonObject, fieldPath);
+      queryResponse.assertFullyRetrieved();
+
+      resultJson.add(schemaFieldName, queryResponse.get());
+    }
+    return resultJson.toString();
+  }
+
+  /**
+   * Get primitive from json by json path or return null if not found.
+   * If element found is not a primitive (object or array) exception is thrown.
+   *
+   * @param path a json path. E.g. "/city/schools/students"
+   * @return a primitive converted to string
+   */
+  @Override
+  public String getPrimitiveByPath(String path) {
+    JSONUtil.JsonQueryResponse queryResponse = JSONUtil.getJsonElementByPath(
+      json, path);
```

Comment:
nit: looks like it could on previous line

Reply:
done